live-site/components/hacking-remaning/HackingRemaining.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from 'react';
+import {
+  StyledHackingRemainingContainer,
+  StyledHackingRemainingHeader,
+  StyledTime
+} from './HackingRemaining.styles';
+
+const HackingRemaining: React.FC = () => {
+  interface TimeLeft {
+    hours: number;
+    minutes: number;
+    seconds: number;
+  }
+
+  const calculateTimeLeft = (): TimeLeft => {
+    const difference = +new Date('2023-02-28T18:20:00+05:30') - +new Date();
+    let timeLeft = { hours: 0, minutes: 0, seconds: 0 };
+
+    if (difference > 0) {
+      timeLeft = {
+        hours: Math.floor(difference / (1000 * 60 * 60)),
+        minutes: Math.floor((difference / 1000 / 60) % 60),
+        seconds: Math.floor((difference / 1000) % 60)
+      };
+    }
+
+    return timeLeft;
+  };
+
+  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft());
+
+  useEffect(() => {
+    setTimeout(() => {
+      setTimeLeft(calculateTimeLeft());
+    }, 1000);
+  });
+
+  return (
+    <StyledHackingRemainingContainer>
+      <StyledHackingRemainingHeader>
+        HACKING REMAINING
+      </StyledHackingRemainingHeader>
+      <StyledTime>
+        <span>{timeLeft.hours}</span>
+        <span>:</span>
+        <span>{timeLeft.minutes}</span>
+        <span>:</span>
+        <span>{timeLeft.seconds}</span>
```

Comment:
To reduce code duplication you can make your timeLeft object look soemthing like 
```
     timeLeft = [
     { timeType: 'hours', value: Math.floor(difference / (1000 * 60 * 60))}.
    { timeType: 'minutes', value: Math.floor((difference / 1000 / 60) % 60)}.
    { timeType: 'seconds', value: Math.floor((difference / 1000) % 60}
      ];
```
and then you can map over this and then just render the span once with timeLeft.value